src/AppInstallerCommonCore/MsixManifest.cpp

```diff
@@ -42,55 +26,66 @@ namespace AppInstaller::Msix
     {
         ComPtr<IAppxManifestPackageId> manifestPackageId;
         THROW_IF_FAILED(m_manifestReader->GetPackageId(&manifestPackageId));
-        return MsixPackageManifestIdentity{ manifestPackageId };
+        return MsixPackageManifestIdentity{ std::move(manifestPackageId) };
     }
 
-    std::optional<OSVersion> MsixPackageManifest::GetMinimumOSVersion() const
+    OSVersion MsixPackageManifest::GetMinimumOSVersion() const
     {
         std::optional<OSVersion> minOSVersion;
         auto targetDeviceFamilies = GetTargetDeviceFamilies();
+
         for (const auto& targetDeviceFamily : targetDeviceFamilies)
         {
-            auto targetDeviceFamilyMinOSVersion = targetDeviceFamily.GetMinVersion();
-            if (!minOSVersion.has_value())
+            auto platform = targetDeviceFamily.GetPlatform();
+            auto minVersion = targetDeviceFamily.GetMinVersion();
+            if (platform == MsixPackageManifestTargetDeviceFamily::Platform::WindowsDesktop)
             {
-                minOSVersion = targetDeviceFamilyMinOSVersion;
+                return minVersion;
             }
-            else if(targetDeviceFamilyMinOSVersion < minOSVersion.value())
+
+            if (platform == MsixPackageManifestTargetDeviceFamily::Platform::WindowsUniversal)
             {
-                minOSVersion = targetDeviceFamilyMinOSVersion;
+                minOSVersion = minVersion;
             }
         }
 
-        return minOSVersion;
+        // If target device family "Windows.Desktop" was not present,
+        // get minimum OS version from "Windows.Universal"
+        if (minOSVersion.has_value())
+        {
+            return minOSVersion.value();
+        }
+
+        // Can't get min OS version because none of the provided
+        // target device family names are supported
+        THROW_HR(E_UNEXPECTED);
```

Comment:
That isn't unexpected, there can be valid MSIX packages with this.  It should be better handled than this will result in.

The two comments on this weren't merged properly 😄 One comment was "this is not optional" and the other was "we should only care about the platforms we support".  Implementing the second comment made the value optional, so it should remain optional on the return type.  And I would change the function name to `GetMinimumOSVersionForSupportedPlatforms`.

Then the caller will need to fail with "No supported platforms" when this value is `null_opt`.

Finally, the function is implemented in a way that will make addition of another supported platform require basically rewriting it.  Why not simply find the minimum version of the supported platforms?

Reply:
Changed the function name, return value to optional, and updated the implementation to allow extending it in the future, but kept the logic to align with this [comment](https://github.com/microsoft/winget-cli/pull/2215#discussion_r890630289). Also added an additional test for "No supported platforms".